ColorzCore/Preprocessor/Directives/PoolDirective.cs

```diff
@@ -15,14 +15,23 @@ class PoolDirective : IDirective
 
         public Maybe<ILineNode> Execute(EAParser p, Token self, IList<IParamNode> parameters, MergeableGenerator<Token> tokens)
         {
+            BlockNode result = new BlockNode();
+
             foreach (List<Token> line in p.PooledLines)
             {
-                tokens.PrependEnumerator(line.GetEnumerator());
+                MergeableGenerator<Token> tempGenerator = new MergeableGenerator<Token>(line);
```

Comment:
I made this mostly because I felt like it was cleaner, not because of any specific thing you mentioned (no offense).

Tho in the end this proved to be useful to fix scope stuff (as I ended up needing to parse those "manually" anyway to be able to pass scope from `AddToPool` to the pooled statements).

Reply:
Use an explicit for loop to avoid the case where AddToPool is nested (eg 
`AddToPool(UNIT whatever AddToPool(REDA whatever); UNIT)`
)
In the interpretation of this, I believe the latter add to pool would get added to the pooled lines, but never iterated over, and then cleared in line 32.